zerver/tests/fixtures/markdown_test_cases.json

```diff
@@ -588,6 +588,16 @@
       "input": ":poop:",
       "expected_output": "<p><span aria-label=\"poop\" class=\"emoji emoji-1f4a9\" role=\"img\" title=\"poop\">:poop:</span></p>"
     },
+    {
+      "name": "emoji_sequence_one",
+      "input": "🤷‍♀️",
+      "expected_output": "<p><span aria-label=\"woman shrugging\" class=\"emoji emoji-1f937-200d-2640\" role=\"img\" title=\"woman shrugging\">:woman_shrugging:</span></p>"
```

Comment:
Thanks for taking this on!

It looks like these two test cases are both about emoji sequences that we know about in our list of Unicode emoji.

In order to fix #11767, it's important that we also handle emoji sequences that we *don't* already know about. The original core problem in #11767 is that when the whole emoji sequence isn't one we recognize, but it has a prefix (often just the first codepoint) that we do recognize, we would corrupt the emoji by transforming just that prefix into our own custom way of describing emoji: https://github.com/zulip/zulip/issues/11767#issuecomment-595462294

So for example until very recently we didn't know about 🐦‍⬛ `:black_bird:`, because it was new in Unicode 15.0. (In fact it looks like on CZO we still don't.) But the emoji exists in people's emoji keyboards on their phones, and can be copy-pasted from elsewhere. If someone typed a "black bird" emoji into a message, Zulip would corrupt it into a mojibake of 🐦"bird" followed by ⬛ "black square". We should instead leave it alone, untouched — then at least if a recipient has the relevant emoji in their own device's font, they'll see it correctly.

Coming up in Unicode 15.1, it looks like there are a bunch of new emoji sequences with novel combinations:
https://emojipedia.org/emoji-15.1
like "bird" + "fire" -> "phoenix", or "lemon" + "green" -> "lime".

Some good test cases would be:
* a sampling of those 15.1 sequences like "phoenix bird", "head shaking horizontally", "person running facing right".
* some fanciful made-up combinations of a random emoji and a colored square or "fire" or "collision" — the sort of ingredients that do seem to get put together into new sequences — as a stand-in for the future sequences that future Unicode versions will add in 2024 and 2025 and beyond.
  * Best if these not only don't yet exist in the standard but seem unlikely to ever do so, as that would defeat the test. For example a good one might be "lemon" + U+1F7EA LARGE PURPLE SQUARE — there probably aren't a lot of purple citrus out there vying to be given the "lime" treatment.
* similarly, some longer fanciful made-up sequences — like "person" + some skin tone + some implement + some motion — taking inspiration from the longer Unicode 15.1 sequences like "person in motorized wheelchair facing right".
  * Again ideally with some silly/incongruous implement and/or motion that makes the combination unlikely to ever make it into the standard.
* a sampling of some sequences involving skin tone. We don't yet *support* skin tone modifiers — that's #992 — but we have not fixed #11767 until we are no longer *corrupting* them and making mojibake out of them.

Reply:
Thanks for this comment! I added a few of these and will add a handful more next time I'm working :)

Reply:
Ok I think I've addressed all your test cases! How's it look?

Reply:
LGTM! Thanks for adding those.